### PR TITLE
core: arm: mutex: simplify code with xsave/xrestore spin lock

### PR DESCRIPTION
**Pull request by MrVan**
cpu_spin_lock_xsave/cpu_spin_unlock_xrestore could be used
to simplify the code.
in __condvar_wait,
```
cpu_spin_unlock(&cv->spin_lock);
cpu_spin_lock(&m->spin_lock);
```
are also replaced with 
```
cpu_spin_unlock_xrestore(&cv->spin_lock, old_itr_status);
old_itr_status = cpu_spin_lock_xsave(&m->spin_lock);
```
Signed-off-by: Peng Fan <peng.fan@nxp.com>

### CODE REVIEW

File: core/arch/arm/kernel/mutex.c [hunk 1]
```diff
@@ -234,17 +226,15 @@ static void __condvar_wait(struct condvar *cv, struct mutex *m,
...
-
 	/* Link this condvar to this mutex until reinitialized */
-	cpu_spin_lock(&cv->spin_lock);
+	old_itr_status = cpu_spin_lock_xsave(&cv->spin_lock);
 	if (cv->m && cv->m != m)
 		panic("invalid mutex");
 
 	cv->m = m;
-	cpu_spin_unlock(&cv->spin_lock);
+	cpu_spin_unlock_xrestore(&cv->spin_lock, old_itr_status);
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/mutex.c]: There's no point in eventually re-enabling interrupts here.

Comment (MrVan) [inline core/arch/arm/kernel/mutex.c]: ok, I'll use the original code here.

**Comment by MrVan** [comment]
@jenswi-linaro Updated, with
```
cpu_spin_unlock_xrestore(&cv->spin_lock, old_itr_status);
old_itr_status = cpu_spin_lock_xsave(&m->spin_lock);
```
back to 
```
cpu_spin_unlock(&cv->spin_lock);
cpu_spin_lock(&m->spin_lock);
```

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by MrVan** [comment]
tags applied.

**Pull request merged**